TypedSyntax/src/show.jl

```diff
@@ -94,17 +94,17 @@ function is_show_annotation(@nospecialize(T); type_annotations::Bool, hide_type_
     return isa(T, Type) && is_type_unstable(T)
 end
 
+extract_inner_type(x) = nothing
+extract_inner_type(::Type{Type{T}}) where T = T
 function type_annotation_mode(node, @nospecialize(T); type_annotations::Bool, hide_type_stable::Bool)
     kind(node) == K"return" && return false, "", "", ""
     type_annotate = is_show_annotation(T; type_annotations, hide_type_stable)
     pre = pre2 = post = ""
     if type_annotate
-        if T isa Type && T <: Type
```

Comment:
Hmm, I guess this should have been `T <: Type{S} where S` or something?

Reply:
`Union{} <: Type{S} where S` is true so I don't think that would have fixed it either.

Reply:
It should be `T isa DataType` or `Base.isType(T)`